File: python/src/cloud/aws/ec2.py
import pulumi_aws as aws


def create_ec2_instance(
    instance_name: str, instance_type: str, ami_id: str
) -> aws.ec2.Instance:
    """Creates an EC2 instance with the given parameters."""
    instance = aws.ec2.Instance(
        instance_name,
        instance_type=instance_type,
        ami=ami_id,
    )
    return instance

File: typescript/src/cloud/aws/ec2.ts
import * as aws from "@pulumi/aws";


export async function createEC2Instance(instanceName: string, instanceType: string, amiId: string): Promise<aws.ec2.Instance> {
    const instance = new aws.ec2.Instance(instanceName, {
        instanceType: instanceType,
        ami: amiId,
        tags: {
            Name: instanceName,
        },
    });

    return instance;
}